### PR TITLE
Task/write out npz rather than xtc

### PR DESCRIPTION
**Pull request by badisa**
Writes out results as NPZ (uncompressed)

Adds tests for the complex RABFE models to verify the change. About 80 seconds, on RTX 2080, will see how long it takes on T4s. Can move to slow tests/ if that seems like a better place.

### CODE REVIEW

File: tests/test_rabfe.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        with temporary_working_dir() as temp_dir:
+            dG, dG_err = model.predict(ordered_params, mol_a, mol_b, core_idxs, complex_x0, complex_box, "prefix")
+            self.assertIsInstance(dG, float)
+            self.assertIsInstance(dG_err, float)
+            created_files = os.listdir(temp_dir)
+            # 3 npz, 1 pdb and 1 npy per mol due to a->b and b->a
+            self.assertEqual(len(created_files), 10)
+            self.assertEqual(len([x for x in created_files if x.endswith(".pdb")]), 2)
+            self.assertEqual(len([x for x in created_files if x.endswith(".npy")]), 2)
+            self.assertEqual(len([x for x in created_files if x.endswith(".npz")]), 6)
```
Comment (maxentile) [inline tests/test_rabfe.py:87]: Nice

File: tests/test_rabfe.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        )
+        mol_a = hif2a_ligand_pair.mol_a
+        mol_b = hif2a_ligand_pair.mol_b
+
+        core_idxs = setup_relative_restraints_by_distance(mol_a, mol_b)
+
+        ref_coords = get_romol_conf(mol_a)
+        mol_coords = get_romol_conf(mol_b) # original coords
+
+        # Use core_idxs to generate
```
Comment (maxentile) [inline tests/test_rabfe.py:60]:
```suggestion
        # Align using core_idxs
```

File: fe/model_rabfe.py [hunk 3]
```diff
@@ -151,38 +151,25 @@ def predict(self,
...
-            # used for debugging for now, try to reproduce mdtraj error
-            # outfile = open("pickle_"+prefix+"_lambda_idx_" + str(lambda_idx) + ".pkl", "wb")
-            # pickle.dump((res.xs, res.boxes, combined_topology), outfile)
-            traj = mdtraj.Trajectory(res.xs, mdtraj.Topology.from_openmm(combined_topology))
-            traj.unitcell_vectors = res.boxes
-            traj.save_xtc("initial_"+prefix+"_lambda_idx_" + str(lambda_idx) + ".xtc")
+            np.savez(
+                f"initial_{prefix}_lambda_idx_{lambda_idx}.npz",
+                xs=res.xs,
+                boxes=res.boxes,
```
Comment (maxentile) [inline fe/model_rabfe.py:166]: Quick clarification: Are these in float32 or float64? (either is fine)

**Review by maxentile — APPROVED** [review]

File: fe/model_rabfe.py (hunk 3 above)
Comment (badisa) [reply fe/model_rabfe.py:166]: These are currently in float32

**Review by proteneer — APPROVED** [review]
lgtm!

**Pull request merged**